week_1/project/week_1_challenge.py

```diff
@@ -60,16 +60,31 @@ def get_s3_data(context):
     return output
 
 
-@op
-def process_data():
-    pass
+@op(
+    config_schema={"nlargest": int},
+    out=DynamicOut(Aggregation),
+    description="Take a list of Stonks and return the N phattest ones.",
+)
+def process_data(context, stocks: List[Stock]):
+    sorted_stocks = sorted(stocks, key=lambda x: x.high, reverse=True)
+    n = context.op_config["nlargest"]
+    top_n = sorted_stocks[:n]
```

Comment:
I think you can add a check to see if n is greater than the number of items.

Reply:
Thanks for the comment! Actually Python is nice about slicing lists - if `n` is larger than the length of the list, it will just go as far as it can:

```python
In [1]: a = [1]
In [2]: a[:10]
Out[2]: [1]
```

So while it won't fail, it's possible that the function wouldn't return what the user was expecting. I'm guessing it might make sense to log a warning that `only 3 were retrieved but user asked for 5`?